constraint/pkg/client/e2e_test.go

```diff
@@ -834,24 +835,103 @@ func TestE2E_Tracing_Unmatched(t *testing.T) {
 	}
 }
 
-// TestE2E_Review_RegoEvaluationMeta tests that we can get stats out of evaluated constraints.
-func TestE2E_Review_RegoEvaluationMeta(t *testing.T) {
+func TestE2E_DriverCfg(t *testing.T) {
+	tests := []struct {
+		name         string
+		statsEnabled bool
+	}{
+		{
+			name:         "disabled",
+			statsEnabled: false,
+		},
+		{
+			name:         "enabled",
+			statsEnabled: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			c := clienttest.New(t)
+
+			_, err := c.AddTemplate(ctx, clienttest.TemplateDeny())
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			_, err = c.AddConstraint(ctx, cts.MakeConstraint(t, clienttest.KindDeny, "foo"))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			obj := handlertest.Review{Object: handlertest.Object{Name: "bar"}}
+
+			rsps, err := c.Review(ctx, obj, drivers.Stats(tt.statsEnabled))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			stats := rsps.StatsEntries
+			if stats == nil && tt.statsEnabled {
+				t.Fatal("got nil stats but stats enabled for Review")
+			} else if len(stats) != 0 && !tt.statsEnabled {
+				t.Fatal("got stats but stats disabled")
+			}
+
+			_, err = c.AddData(ctx, &obj.Object)
```

Comment:
Why is AddData() called? What is it testing?

Reply:
this shouldn't have been there, thanks for the spot!